identity-service/src/routes/trackListens.js

```diff
@@ -15,6 +15,107 @@ let oneWeekInMs = oneDayInMs * 7
 let oneMonthInMs = oneDayInMs * 30
 let oneYearInMs = oneMonthInMs * 12
 
+const parseTimeframe = (inputTime) => {
+  switch (inputTime) {
+    case 'day':
+    case 'week':
+    case 'month':
+    case 'year':
+    case 'millennium':
+      break
+    default:
+      inputTime = undefined
+  }
+
+  // Allow default empty value
+  if (inputTime === undefined) {
+    inputTime = 'millennium'
+  }
+  return inputTime
+}
+
+const getTrackListens = async (
+  idList,
+  timeFrame = undefined,
+  startTime = undefined,
```

Comment:
i think this undefined pattern might be unnecessary based on the code below actually